dbtwiz/generate/model.py

```diff
@@ -0,0 +1,156 @@
+import os
+import yaml
+
+from dbtwiz.interact import (
+    input_text,
+    select_from_list,
+    multiselect_from_list,
+    autocomplete_from_list,
+)
+
+from dbtwiz.logging import info, warn, error, fatal
+
+from dbtwiz.model import (
+    Group,
+    Project,
+    layer_choices,
+    materialization_choices,
+    access_choices,
+    domains_for_layer,
+    model_base_path,
+)
+
+
+def generate_model(quick: bool):
+    """Generate new dbt model"""
+
+    project = Project()
+
+    try:
+        layer = select_from_list(
+            "Select model layer",
+            layer_choices())
+
+        domain = autocomplete_from_list(
+            "Which domain does your model belong to",
+            domains_for_layer(layer),
+            must_exist=False,
+            allow_blank=False)
+
+        while True:
+            name = input_text(
+                "What is the name of your model")
+            base_path = model_base_path(layer, domain, name)
+            sql_path = base_path.with_suffix(".sql")
+            yml_path = base_path.with_suffix(".yml")
+            if sql_path.exists() or yml_path.exists():
+                error("A model with that name already exists, please choose another.")
+            else:
+                break
+
+        description = input_text(
+            "Give a short description of your model and its purpose")
+
+        if not quick:
+            group = autocomplete_from_list(
+                "Which group does your model belong to",
+                Group().choices(),
+                must_exist=True,
+                allow_blank=True)
+
+            access = select_from_list(
+                "What is the access level for this model",
+                access_choices())
+
+            materialization = select_from_list(
+                "How should your model be materialized",
+                materialization_choices())
+
+            teams = multiselect_from_list(
+                "Which team(s) should be responsible for this model",
```

Comment:
You have the same issue that I had originally here. If the use simply presses "Enter" without having selected one of the options, there is no error message and the option that was selected isn't used.
In my suggested PR, this was handled by allowing Enter to select the currently selected option.

`? Which team(s) should be responsible for this model: done`

Reply:
Fixed by adding validation, but don't need the while-loop then as there is no way to return from it without selecting or cancelling the whole process.